managedtenants/bundles/ci.py

```diff
@@ -306,32 +286,33 @@ def __init__(self, args):
     def run(self):
         target_addons = self._get_target_addons()
 
-        for addon in target_addons:
-            bundle_images, latest_version = build_and_push_addon_bundles(
-                bundle_builder=self._bundle_builder(addon)
-            )
+        for addon_dir in target_addons:
+            addon_bundles = AddonBundles(addon_dir)
+            bundles = addon_bundles.get_all_bundles()
 
-            index_builder = self._index_builder(addon)
-            index_image = index_builder.build_push_index_image(
-                bundle_images=bundle_images,
-                hash_string=get_short_hash(),
+            bundles = self.bundle_builder.build_and_push_all(
+                bundles, get_short_hash()
+            )
+            index_image = self.index_builder.build_and_push(
+                bundles, get_short_hash()
```

Comment:
This is the pipelining bit I mentioned above. It's useful if we skip failed bundles, but otherwise it just adds another check back to `BundleBuilder` to reason about which bundles are being included in the index.

Reply:
removed pipelining